pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageImpl.java

```diff
@@ -64,6 +64,7 @@
     private ByteBuf payload;
 
     private Schema<T> schema;
+    private SchemaInfo schemaInfo;
```

Comment:
For the schema replication, we don't need to convert schemaInfo to schema for the local topic and then convert the schema to the schemaInfo for the remote topic. Or we only need to pass the schemaInfo to the message which need to replicate to the remote cluster.

Reply:
We already have a schema, but we added a schemaInfo, which will make other developers very confused. If we can't reuse existing attributes, we can at least do this:
1) Make the new properties easier to understand, such as: naming modification, or adding comments, so that other developers can know schemaInfo only for replicator
2) Independent get/set

Reply:
I second @315157973 concern.
We already have "getReaderSchema", why cannot we use that ?